hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/impl/BlockStreamManagerImpl.java

```diff
@@ -338,8 +341,10 @@ public synchronized void accept(@NonNull final byte[] message, @NonNull final by
                     .siblingHashes(siblingHashes.stream().flatMap(List::stream).toList());
             block.writer()
                     .writeItem(BlockItem.PROTOBUF.toBytes(
-                            BlockItem.newBuilder().blockProof(proof).build()))
-                    .closeBlock();
+                            BlockItem.newBuilder().blockProof(proof).build()));
+            if (streamWriterType == BlockStreamWriterMode.FILE) {
+                block.writer().closeBlock();
+            }
```

Comment:
This feels like a weird place to put this since it only applies to files. Maybe we have a method in this class that the file writer can override instead

Reply:
Method `closeBlocks()` in BlockItemWriter is implemented by both writers (file and grpc). We just want to call them on different occassions. The file writer should be closed when we write the `BlockProof` which is the last `BlockItem` in every written Block. In the case of the grpc writer, we want to call `closeBlock()` whenever we receive `EndOfStream` object from the block node which indicates the stream ended, not immediately on streaming the `BlockProof`